### PR TITLE
[DD4hep] Keep mag. field geom. builder shape lengths same with DD4hep units change

### PR DESCRIPTION
**Pull request by cvuosalo**
DD4hep currently follows the ROOT units convention (cm = 1). In January it will be switched to follow the Geant4 units convention (mm = 1). This PR changes the magnetic field geometry builder code so shape sizes will stay the same before and after the change.

#### PR validation:

The output of the `MagneticField/GeomBuilder/test/python/testMagGeometry.py` script (with debugBuilder = cms.untracked.bool(True)) before and after the PR was checked and seen to be identical.

No backport is needed.

### CODE REVIEW

**Comment by cvuosalo** [comment]
@cmsbuild please test

File: MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc [hunk 1]
```diff
@@ -47,62 +51,71 @@ volumeHandle::volumeHandle(const cms::DDFilteredView &fv, bool expand2Pi, bool d
...
       double deltaPhi = tubs.endPhi() - startPhi;
       buildTubs(zhalf, rIn, rOut, startPhi, deltaPhi);
     } break;
     case DDSolidShape::ddpseudotrap: {
       vector<double> d = solid.parameters();
+      if (d.size() >= 7)
+        LogTrace("MagGeoBuilder") << " Pseudo trap params raw =  " << d[0] << ", " << d[1] << ", " << d[2] << ", "
+                                  << d[3] << ", " << d[4] << ", " << d[5] << ", " << d[6];
+
+      transform(d.begin(), --(d.end()), d.begin(), [](double val) { return val / dd4hep::cm; });
```
Comment (jpata) [inline MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc:102]: is it intented to transform all inputs **except** the last one in the list? Given the debugging comments around this line, I suppose the last parameter d[6] has some special meaning, which I was not able to dig up immediately. Perhaps a code comment could clarify things for future readers?

Comment (namapane) [reply MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc:102]: yes, indeed d[6] is to be interpreted as a bool, cf:
https://github.com/cms-sw/cmssw/blob/134ed3345e18f52b5c8e50e3217d326b760b6402/MagneticField/GeomBuilder/src/buildPseudoTrap.icc#L16
That should be clear from the signature of that function, but I agree a comment here would not harm.

Comment (jpata) [reply MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc:102]: Yes, please put a comment here, to make a bit clearer that the transform relies on the special signature of the `buildPseudoTrap` function.

Please also change the following to be even more explicit

```C++
buildPseudoTrap(d[0], d[1], d[2], d[3], d[4], d[5], static_cast<bool>(d[6]));
```

I remark here that doing bool->double->bool seems a bit odd in general, but I don't know enough about this code to suggest a better alternative right now.

Comment (cvuosalo) [reply MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc:102]: Sorry, I should have put a comment here. The pseudo-trapezoid is a special case that is only used for legacy configurations. I will add clarifications.

**Comment by jpata** [comment]
@cmsbuild please test

**Comment by jpata** [comment]
+reconstruction
- technical, readiness for using cm units in `MagneticField/GeomBuilder`
- no changes expected or observed

**Comment by qliphy** [comment]
+1

**Pull request merged**